v2/runner/collector.go

```diff
@@ -61,23 +61,29 @@ func StartCollectorWithContext(ctx context.Context, collector plugin.Collector,
 
 func startCollector(ctx context.Context, collector types.Collector) {
 	var err error
-
 	var opt *plugin.Options
+	var ctxLog logrus.FieldLogger
+
 	inprocPlugin, inProc := collector.Unwrap().(inProcessPlugin)
 	if inProc {
 		opt = inprocPlugin.Options()
-
-		logger := inprocPlugin.Logger()
-		ctx = log.ToCtx(ctx, logger)
+		ctxLog = inprocPlugin.Logger()
 
 		switch collector.Type() {
 		case types.PluginTypeCollector:
 			collector = types.NewCollector(collector.Name(), collector.Version(), inprocPlugin.Unwrap().(plugin.Collector))
 		case types.PluginTypeStreamingCollector:
 			collector = types.NewStreamingCollector(collector.Name(), collector.Version(), inprocPlugin.Unwrap().(plugin.StreamingCollector))
 		}
+	} else {
+		ctxLog = logrus.WithFields(logrus.Fields{
```

Comment:
not sure if it is needed. For standard plugins, snap will add plugin-name to each log line ? Did you check how this will look ?

Reply:
hah, good point 👍

Reply:
ok, will close this one without merge